WordPress/Classes/Networking/BlogServiceRemoteREST.m

```diff
@@ -210,20 +210,22 @@ - (NSDictionary *)mapPostFormatsFromResponse:(id)response
 
 - (RemoteBlogSettings *)remoteBlogSettingFromJSONDictionary:(NSDictionary *)json
 {
-    RemoteBlogSettings *remoteSettings = [[RemoteBlogSettings alloc] init];
+    NSDictionary *rawSettings = [json dictionaryForKey:@"settings"];
+    
+    RemoteBlogSettings *remoteSettings = [RemoteBlogSettings new];
     
     remoteSettings.name = [json stringForKey:@"name"];
     remoteSettings.desc = [json stringForKey:@"description"];
-    
-    if (json[@"settings"][@"default_category"]) {
-        remoteSettings.defaultCategory = [json numberForKeyPath:@"settings.default_category"];
-    } else {
-        remoteSettings.defaultCategory = @(PostCategoryUncategorized);
-    }
-    if ([json[@"settings"][@"default_post_format"] isEqualToString:@"0"]) {
+    remoteSettings.defaultCategory = [rawSettings numberForKey:@"default_category"] ?: @(PostCategoryUncategorized);
+
+    // Note:
+    // YES, the backend might send '0' as a number, OR a string value.
+    //
```

Comment:
I'd consider adding a link to the ticket you're solving in the comments - for extra reinforcement of why this is being done. (https://github.com/wordpress-mobile/WordPress-iOS/issues/4187)